GNUmakefile

```diff
@@ -51,13 +54,18 @@ endif
 
 INSTALL_NAME_TOOL := install_name_tool
 
-STORE_DIR        := store
-BINDIST_BASE_DIR := out/bindist
+STORE_DIR        := store/$(ARTIFACT)
+BINDIST_BASE_DIR := out/bindist/$(ARTIFACT)
 BINDIST_OUT_DIR  := $(BINDIST_BASE_DIR)/haskell-language-server-$(HLS_VERSION)
 
-CABAL_ARGS         ?= --store-dir=$(ROOT_DIR)/$(STORE_DIR)
-CABAL_INSTALL_ARGS ?= --disable-tests --disable-profiling -O2 --overwrite-policy=always --install-method=copy
-CABAL_INSTALL      := $(CABAL) $(CABAL_ARGS) v2-install
+CABAL_BASE_ARGS         ?= --store-dir=$(ROOT_DIR)/$(STORE_DIR)
+CABAL_ARGS              ?= --disable-tests --disable-profiling -O2
+CABAL_INSTALL_ARGS      ?= --overwrite-policy=always --install-method=copy
+CABAL_INSTALL           := $(CABAL) $(CABAL_BASE_ARGS) v2-install
+PROJECT_FILE            := cabal.project
```

Comment:
Again I'm confused about why it's useful to do `cabal --project-file cabal.project` when that's the default?

Reply:
Explicit is sometimes better. You can pass a different project file without messing with the make rules.